test/private/calculateMD5Hash.m

```diff
@@ -14,19 +14,18 @@
             % is a  cryptographic hash, even though its security has been
             % broken. Instead we make use of the Java libraries.
             % Unless the "-nojvm" flag is specified, this should work well.
-
-            % Java only has reliable support for absolute paths.
-            absoluteFilename = fullfile(pwd, filename);
-
-            % Based on code by St�phane Pinchaux and Bastian Ebeling that can be
-            % found at <http://stackoverflow.com/questions/12140458/>.
-
-            fis = java.io.FileInputStream(java.io.File(absoluteFilename));
+            
+            % Get message digest instance
             MD5 = java.security.MessageDigest.getInstance('MD5');
-            dis = java.security.DigestInputStream(fis, MD5);
-
-            while(dis.read() ~= -1), end; % read the whole file
-
+                        
+            % Open the file
+            fid     = fopen(filename, 'r');
+            cleanup = onCleanup(@()fclose(fid));
+            
+            % Read file into a stream of uint8
+            stream = fread(fid, '*uint8');
```

Comment:
Technically this is not a stream but an array. Anyhow, if there is a reference that you've used (except for the Java documentation), please mention it in the comments.